java_src/foam/dao/MDAO.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package foam.dao;
+
+import foam.core.ClassInfo;
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+import foam.dao.index.AltIndex;
+import foam.dao.index.Index;
+import foam.dao.index.Plan;
+import foam.dao.index.SelectPlan;
+import foam.dao.index.TreeIndex;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.mlang.predicate.True;
+
+public class MDAO extends AbstractDAO {
+  
+  protected ClassInfo of_;
+  protected Index index;
+  protected Object state;
+  
+  public MDAO(ClassInfo of_) {
+    this.of_ = of_;
+    state = null;
+    index = new AltIndex(new TreeIndex((PropertyInfo) this.of_.getAxiomByName("id")));
+  }
+  
+  public void addIndex(PropertyInfo prop) {
+    if ( ! (index instanceof AltIndex) ) {
```

Comment:
You don't need this check since you already set index to an AltIndex in the constructor. You could also just make index_ be of type AltIndex so you don't have to cast it since you know that's always the case.